inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,348 @@
+            else:
+                if self.match_type == 'exact' and record[key] == self.value_to_check:
+                    del record[key]
+
+                elif self.match_type == 'regex' and re.search(
+                        self.value_to_check, record[key]):
+                            del record[key]
+                elif self.match_type == 'contains' and self.value_to_check in record[key]:
+                    del record[key]
+                self.changed = True
+                return
+        else:
+            if isinstance(record[key], list):
+                for array_record in record[key]:
+                    self.apply(array_record, schema, position + 1, conditions_passed)
+            else:
+                self.apply(record[key], schema, position + 1, conditions_passed)
+        if isinstance(record[key], list):
+            record[key] = [item for item in record[key] if item not in [{}, '', []]]
+        if record[key] in [{}, '', []]:
+            del record[key]
+
+
+class Update(Action):
+
+    def apply(self, record, schema=None, position=0, conditions_passed=0):
+        """Recursive function to update a record primitive key."""
+        conditions_passed, key, condition_failed = self.process_keypath(record, position, conditions_passed)
+        if condition_failed:
+            return
+        if not record.get(key):
+            return
+        if position == len(self.keypath) - 1:
+            if isinstance(record[key], list):
+                if self.match_type == 'regex':
+                    record[key] = [self.value if re.search(
+                        self.value_to_check, x) else x for x in record[key]]
+
+                elif self.match_type == 'exact':
+                    record[key] = [self.value if x == self.value_to_check else x for x in record[key]]
+
+                elif self.match_type == 'contains':
+                    record[key] = [self.value if self.value_to_check in x else x for x in record[key]]
+                self.changed = True
+            else:
+                if self.match_type == 'exact' and record[key] == self.value_to_check:
+                    record[key] = self.value
+                if self.match_type == 'regex' and re.search(
+                        self.value_to_check, record[key]):
+                            record[key] = self.value
+                elif self.match_type == 'contains' and self.value_to_check in record[key]:
+                    record[key] = self.value
+                self.changed = True
+                return
+        else:
+            if isinstance(record[key], list):
+                for array_record in record[key]:
+                    self.apply(array_record, schema, position + 1, conditions_passed)
+            else:
+                self.apply(record[key], schema, position + 1, conditions_passed)
+
+
+def create_schema_record(schema, path, value):
+    """Object creation in par with the schema."""
+    record = {}
+    temp_record = record
```

Comment:
I don't know python internals but this might end up `record` and `temp_record` having same references!
Maybe what you want actually `temp_record = {}`

Reply:
now that checked again, you actually don't even use record at all, but modify `temp_record` which reflects on `record` then return `record`. So just remove `temp_record`